cpp/oneapi/dal/algo/pca/backend/cpu/train_kernel_svd.cpp

```diff
@@ -38,13 +39,20 @@ template <typename Float, daal::CpuType Cpu>
 using daal_pca_svd_kernel_t = daal_pca::internal::
     PCASVDBatchKernel<Float, daal_pca::interface3::BatchParameter<Float, daal_pca::svdDense>, Cpu>;
 
+template <typename Float>
+inline auto get_normalization_algorithm() {
+    using normalization_alg_t = daal::algorithms::normalization::zscore::
+        Batch<Float, daal::algorithms::normalization::zscore::defaultDense>;
+    return daal::services::SharedPtr<normalization_alg_t>{ new normalization_alg_t{} };
```

Comment:
It cannot be deduced, because `SharedPtr<T>` constructor accepts pointer of different type `U` from which `T` cannot be deduced without deduction guide.